File: Example/cenarius/src/main/java/com/m/cenarius/Web/Interceptor/RouteInterceptor.java
package com.m.cenarius.Web.Interceptor;

import android.content.Context;

import com.m.cenarius.Route.Route;
import com.m.cenarius.Utils.UrlUtil;

/**
 * Created by m on 2017/5/10.
 */

public class RouteInterceptor implements InterceptorAdapter {

    @Override
    public boolean perform(String url, Context controller) {
        if ("cenarius".equals(UrlUtil.getScheme(url)) && "route".equals(UrlUtil.getHost(url))) {
            Route.open(url, controller);
            return true;
        }
        return false;
    }
}

File: cenarius/src/main/java/com/m/cenarius/Web/Interceptor/RouteInterceptor.kt
package com.m.cenarius.Web.Interceptor

import android.content.Context
import android.text.TextUtils
import com.m.cenarius.Extension.*

import com.m.cenarius.Route.Route

/**
 * Created by m on 2017/5/10.
 */

class RouteInterceptor : InterceptorAdapter {

    override fun perform(url: String, controller: Context): Boolean {
        if (TextUtils.equals(url.getScheme(), Interceptor.scheme) && TextUtils.equals(url.getHost(), "route")) {
            Route.open(url, controller)
            return true
        }
        return false
    }
}
